### PR TITLE
MEN-3771 support multiple device types

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
quick fix -- I am not sure it will work, @kacf and @mzedel fyi.

File: cmd/single-file.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+ "strings"
```
Comment (kacf) [inline cmd/single-file.go:24]: go fmt failure.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,4 @@
-create-artifact-worker
\ No newline at end of file
+create-artifact-worker
+create-artifact
+.idea
+.*.swp
```
Comment (kacf) [inline .gitignore:4]: Please put in a separate commit (important when cherry-picking).

File: cmd/single-file.go [hunk 3]
```diff
@@ -256,14 +257,40 @@ func (c *SingleFileCmd) Run() error {
...
-		"-n", c.ArtifactName,
-		"-t", c.DeviceType,
+	deviceTypes := strings.Split(c.DeviceType, ",")
+	/*  the artifactGenArguments array length comes from:
+	    "-n", c.ArtifactName,
 		"-d", c.DestDir,
 		"-o", outfile,
 		downloadFile,
+	*/
+	artifactGenArguments := make([]string, 2*len(deviceTypes)+2*3+1)
```
Comment (kacf) [inline cmd/single-file.go:267]: Suggestion: Drop the allocation, and then use `artifactGenArguments = append(artifactGenArguments, "-n", c.ArtifactName)` pattern below instead, which is less error prone, and requires fewer lines.

And in addition, if you really want to preallocate, use `make([]string, 0, 2*len(deviceTypes)+2*3+1)`. I would leave that out though, this part is not performance critical, so better to not have to worry about the size calculation.

File: cmd/single-file.go [hunk 4]
```diff
@@ -256,14 +257,40 @@ func (c *SingleFileCmd) Run() error {
 
 	mlog.Verbose("generating output artifact %s", outfile)
 
-	// run gen script
-	cmd := exec.Command(
-		"/usr/bin/single-file-artifact-gen",
-		"-n", c.ArtifactName,
-		"-t", c.DeviceType,
+	deviceTypes := strings.Split(c.DeviceType, ",")
```
Comment (kacf) [inline cmd/single-file.go:260]: In spirit, this looks correct, but I'll leave for someone else to evaluate whether this is correct place to do it.

File: cmd/single-file.go (hunk 3 above)
Comment (alfrunes) [inline cmd/single-file.go:267]: can be written a little more compactly:
```
args := []string{
  "-n", c.ArtifactName,
  "-d", c.DestDir,
  "-o", outfile,
}
for _, devType := range deviceTypes {
  args = append(args, "-t", devType)
}
args = append(args, outfile)
exec.Command(...
```

File: cmd/single-file.go (hunk 1 above)
Comment (alfrunes) [inline cmd/single-file.go:24]: `go fmt`

**Comment by tranchitella** [comment]
@merlin-northern alternative PR is here: https://github.com/mendersoftware/create-artifact-worker/pull/34
I've tested it locally and it works fine.

**Comment by tranchitella** [comment]
Superseded by https://github.com/mendersoftware/create-artifact-worker/pull/34
